pandas/tests/indexes/datetimelike.py

```diff
@@ -38,3 +40,31 @@ def test_view(self):
         i_view = i.view(self._holder)
         result = self._holder(i)
         tm.assert_index_equal(result, i_view)
+
+    def test_add_timedelta(self):
+        # GH 17558
+        # Check that tz-aware DatetimeIndex + np.array(dtype="timedelta64")
+        # and DatetimeIndex + TimedeltaIndex work as expected
+        idx = self.create_index()
+        idx.name = "x"
+        if isinstance(idx, pd.DatetimeIndex):
+          idx = idx.tz_localize("US/Eastern")
+
+        expected = idx + np.timedelta64(1, 'D')
+        tm.assert_index_equal(idx, expected - np.timedelta64(1, 'D'))
```

Comment:
just to explain the awkwardness here: was trying to figure out how to demonstrate that `expected` actually has the properties we expect, e.g. has `name = "x"` and is a day ahead of `idx`